crates/ruff_linter/src/rules/flake8_return/rules/function.rs

```diff
@@ -688,3 +726,34 @@ pub(crate) fn function(checker: &mut Checker, body: &[Stmt], returns: Option<&Ex
         }
     }
 }
+
+fn remove_else(checker: &mut Checker, elif_else: &ElifElseClause) -> Result<Fix> {
+    if elif_else.test.is_some() {
+        // it's an elif, so we can just make it an if
+
+        // delete "el" from "elif"
+        Ok(Fix::safe_edit(Edit::deletion(
+            elif_else.start(),
+            elif_else.start() + TextSize::from(2),
+        )))
+    } else {
+        let else_line_range = checker.locator().full_line_range(elif_else.start());
```

Comment:
I don't expect this to exist in practice but assuming that the `else` header ends on the same line as the `else` keyword is not always true. For example if you have.

```python
if True:
    return

else\
    :\
    # comment
    pass
```

it then keeps `:\\n   # comment` where it should not. We could use `SimpleTokenizer` to detect this but I don't think it's worth doing. 

However, I think it would be good to add a few more tests that demonstrate how the fix deals with comments in different positions. 

```python
if True:
    return
else:
    # comment
    pass
```

```python
if True:
    return
else: # comment
    pass
```

Reply:
I've tweaked it to work with that line continuation that I didn't account for, and the code now keeps all comments after the else's `:` 😄